k2/csrc/array_test.cu

```diff
@@ -108,6 +108,25 @@ void TestArray1() {
     }
   }
 
+  {
+    // created with Array1(ContextPtr, int32_t size, Callable &&callable)
+    auto lambda_set_values = [=] __host__ __device__(int32_t i) -> T {
+      return i * i;
+    };
+    Array1<T> array(context, 5, lambda_set_values);
+    ASSERT_EQ(array.Dim(), 5);
```

Comment:
The function `operator[](int32_t i)` of `Array1` is very helpful for testing. You can use
```cpp
for (int32_t i = 0; i != array.Dim(); ++i) {
  EXPECT_EQ(array[i], i * i);
}
 ```
This requires far less code.

Reply:
Oh, yes, as we have implemented `operator[](int32_t)`. Updated other tests as well. Thanks!